### PR TITLE
🐛 Don't render CTA button when there are no shopping tags

### LINKED ISSUE #37004: [amp story shopping] Do not render CTA if no product tags are on page
### Description

The CTA button should not render when there are no product tags on the page.

### PR DESCRIPTION
**Pull request by jshamble**
Doesn't render the cta button when there are no shopping tag elements.

closes #37004

Before            |  After
:-------------------------:|:-------------------------:
Screen Shot 2021-12-13 at 4 49 20 PM  |  Screen Shot 2021-12-13 at 4 53 53 PM

### CODE REVIEW

File: examples/amp-story/amp-story-shopping.html [hunk 1]
```diff
@@ -103,6 +103,45 @@
...
+          <amp-img layout="fill" src="/examples/visual-tests/amp-story/img/shopping/bg.png"></amp-img>
+        </amp-story-grid-layer>
+        <amp-story-grid-layer template="vertical">
+        </amp-story-grid-layer>
+        <amp-story-shopping-attachment></amp-story-shopping-attachment>
+      </amp-story-page>
+      <amp-story-page id="remote-with-product">
+        <!--
+          Example of:
+          CTA Button  redering due to remote product tag existing, but no inline.
```
Comment (processprocess) [inline examples/amp-story/amp-story-shopping.html:128]: Nit: Remove extra space.

File: extensions/amp-story/1.0/amp-story-open-page-attachment.js [hunk 2]
```diff
@@ -191,24 +191,27 @@ const renderInlineUi = (pageEl, attachmentEl) => {
...
-        {makeImgElWithBG('cta-image')}
-        {makeImgElWithBG('cta-image-2')}
-        <div class="i-amphtml-story-inline-page-attachment-arrow"></div>
-      </div>
-      {openLabel !== 'none' && (
-        <span class="i-amphtml-story-page-attachment-label">{openLabel}</span>
-      )}
-    </a>
-  );
+  if (pageEl.getElementsByTagName('amp-story-shopping-tag').length === 0) {
```
Comment (processprocess) [inline extensions/amp-story/1.0/amp-story-open-page-attachment.js:194]: This breaks attachments.
Please check `/examples/amp-story/attachment.html`
This exception should only run for the shopping experience.

File: extensions/amp-story/1.0/amp-story-open-page-attachment.js [hunk 3]
```diff
@@ -191,24 +191,27 @@ const renderInlineUi = (pageEl, attachmentEl) => {
...
+    return <div></div>;
```
Comment (processprocess) [inline extensions/amp-story/1.0/amp-story-open-page-attachment.js:195]: We should never add anything extra to the dom. Performance needs to always be top of mind.
We may be able to make this exception earlier.

There may be a way to do this from amp-story-page. [look here](https://github.com/ampproject/amphtml/blob/main/extensions/amp-story/1.0/amp-story-page.js#L1575-L1582).

**Review by processprocess — CHANGES_REQUESTED** [review]

File: examples/visual-tests/amp-story/amp-story-shopping.html [hunk 4]
```diff
@@ -103,6 +103,45 @@
...
+            {
+                "items" : []
+              }
+          </script>
+        </amp-story-shopping-config>
+        <amp-story-grid-layer template="fill">
+          <amp-img layout="fill" src="/examples/visual-tests/amp-story/img/shopping/bg.png"></amp-img>
+        </amp-story-grid-layer>
+        <amp-story-grid-layer template="vertical">
+          <amp-story-shopping-tag data-tag-id="city-pop" ></amp-story-shopping-tag>
```
Comment (coreymasanto) [inline examples/visual-tests/amp-story/amp-story-shopping.html:141]: Out of curiosity, what does the product tag look like when the `data-tag-id` does not exist? I'm imagining that no tag ends up being displayed

File: extensions/amp-story/1.0/amp-story-page.js [hunk 5]
```diff
@@ -1577,7 +1577,11 @@ export class AmpStoryPage extends AMP.BaseElement {
     const attachmentEl = this.element.querySelector(
       'amp-story-page-attachment, amp-story-page-outlink, amp-story-shopping-attachment'
     );
-    if (!attachmentEl) {
+
+    if (
+      !attachmentEl ||
+      this.element.getElementsByTagName('amp-story-shopping-tag').length === 0
+    ) {
```
Comment (coreymasanto) [inline extensions/amp-story/1.0/amp-story-page.js:1584]: This means that the open attachment UI will not render if `(no attachment element exists) OR (no shopping tags exist)`. Rather, you'll want the condition to be something more like `(no attachment element exists) OR (a shopping attachment exists AND no shopping tags exist)`

File: examples/visual-tests/amp-story/amp-story-shopping.html (hunk 4 above)
Comment (jshamble) [reply examples/visual-tests/amp-story/amp-story-shopping.html:141]: Yes, that's right no tags end up being displayed.

File: extensions/amp-story/1.0/amp-story-page.js (hunk 5 above)
Comment (jshamble) [reply extensions/amp-story/1.0/amp-story-page.js]: Sounds good to me, I added in the additional logic statement.

File: extensions/amp-story/1.0/amp-story-page.js [hunk 6]
```diff
@@ -1577,7 +1577,14 @@ export class AmpStoryPage extends AMP.BaseElement {
     const attachmentEl = this.element.querySelector(
       'amp-story-page-attachment, amp-story-page-outlink, amp-story-shopping-attachment'
     );
-    if (!attachmentEl) {
+
+    if (
+      !attachmentEl ||
+      (attachmentEl &&
```
[the remark this answers is not in the document]
Comment (jshamble) [reply extensions/amp-story/1.0/amp-story-page.js:1583]: Sounds good Corey, I will add the before and images as well to the description, and will do so for future css visual PRs.

**Review by gmajoulet — APPROVED** [review]

**Pull request merged**